iceoryx_utils/include/iceoryx_utils/internal/lockfree_queue/index_queue.inl

```diff
@@ -0,0 +1,274 @@
+namespace iox
+{
+template <uint64_t Capacity, typename ValueType>
+IndexQueue<Capacity, ValueType>::IndexQueue(ConstructEmpty_t)
+    : m_readPosition(Index(Capacity))
+    , m_writePosition(Index(Capacity))
+{
+}
+
+template <uint64_t Capacity, typename ValueType>
+IndexQueue<Capacity, ValueType>::IndexQueue(ConstructFull_t)
+    : m_readPosition(Index(0))
+    , m_writePosition(Index(Capacity))
+{
+    for (uint64_t i = 0; i < Capacity; ++i)
```

Comment:
We have to write ```uint64_t i = 0u``` otherwise stupid tools will babble about it

Reply:
yeah, I am sure if the ranges and signs fit it is no problem, I am not sure if cast overhead is introduced. For integral template types this is especially problematic...we do not know the instantiated type and length there (do we need 0ull, or 0u or whatever).